db/view/view.cc

```diff
@@ -2471,6 +2472,76 @@ future<> view_builder::do_build_step() {
     });
 }
 
+future<> view_builder::migrate_to_v2(db::system_keyspace& sys_ks, cql3::query_processor& qp, service::raft_group0_client& group0_client, abort_source& as, service::group0_guard guard) {
+    // TODO MICHAEL
+    // currently we don't handle concurrent writes to the table while we migrate it,
+    // and we may lose writes.
+    // maybe we can wait until there are no view builds in progress and assume no new
+    // view builds start while we migrate.
+    inject_failure("view_builder_migrate_to_v2");
+
+    auto schema = qp.db().find_schema(db::system_distributed_keyspace::NAME, db::system_distributed_keyspace::VIEW_BUILD_STATUS);
+    auto cl = db::consistency_level::QUORUM; // TODO MICHAEL ?
```

Comment:
I think there is a difference from the service level migration, that the service level migration is part of the raft upgrade procedure and I think it could assume all nodes are up. But here I start the migration when the feature is enabled, not necessarily having all nodes up.
I think maybe I should add a condition to start the migration when all nodes are up and everything is stable, and then I can also change the CL to ALL.
what do you think?

Reply:
ok I changed it to CL ALL, or less if there are less than 3 nodes, like in service levels.
I'm still wondering if it's possible we try to migrate while one node is down and should I try to avoid it?

Reply:
This migration will run right after a cluster feature is enabled, and enabling a feature requires all nodes in the cluster to be up. I think it is a reasonable assumption.

Reply:
I do see in my tests that the migration could fail sometimes because cl ALL is not achieved, but maybe it's just a short temporary failure.
I think what happens is we upgrade the last node, then the feature is enabled and the migration is started but before the last node is ready for the query.